guardrails/guard.py

```diff
@@ -256,8 +257,18 @@ def _fill_validator_map(self):
                     for v in entry
                     if (
                         v.rail_alias == ref.id
-                        and v.on_fail_descriptor == ref.on_fail
-                        and v.get_args() == ref.kwargs
+                        and (
```

Comment:
This is pretty tough to read as a list comprehension so I would argue it could be simpler for maintainability as a loop, but otherwise PR LGTM

Reply:
Fair, I originally had it as a filter which was even worse on the eyes and I believe more expensive since you had to cast it to a list afterwards.  I can swap this to a for-loop instead; should be equivalent.